pkg/analyze/velero.go

```diff
@@ -0,0 +1,631 @@
+			results = append(results, &AnalyzeResult{
+				Title:   "Velero Schedules",
+				IsPass:  true,
+				Message: fmt.Sprintf("Found %d schedules", len(schedules)),
+			})
+		}
+	}
+	return results
+}
+
+func analyzeVolumeSnapshotLocations(volumeSnapshotLocations []*velerov1.VolumeSnapshotLocation) []*AnalyzeResult {
+	results := []*AnalyzeResult{}
+	failures := 0
+	if len(volumeSnapshotLocations) > 0 {
+		for _, volumeSnapshotLocation := range volumeSnapshotLocations {
+			if volumeSnapshotLocation.Status.Phase == velerov1.VolumeSnapshotLocationPhaseUnavailable {
+				result := &AnalyzeResult{
+					Title: fmt.Sprintf("Volume Snapshot Location %s", volumeSnapshotLocation.Name),
+				}
+				result.IsFail = true
+				result.Message = fmt.Sprintf("Volume Snapshot Location %s phase is %s", volumeSnapshotLocation.Name, volumeSnapshotLocation.Status.Phase)
+				results = append(results, result)
+				failures++
+			}
+		}
+		if failures == 0 {
+			results = append(results, &AnalyzeResult{
+				Title:   "Velero Volume Snapshot Locations",
+				IsPass:  true,
+				Message: fmt.Sprintf("Found %d volume snapshot locations", len(volumeSnapshotLocations)),
+			})
+		}
+	}
+
+	return results
+}
+
+func analyzeLogs(logs map[string][]byte) []*AnalyzeResult {
+	results := []*AnalyzeResult{}
+	if len(logs) > 0 {
+		for _, logBytes := range logs {
+			logContent := string(logBytes)
+			result := &AnalyzeResult{
+				Title: fmt.Sprintf("Velero logs for pod [node-agent]"),
+			}
+			if strings.Contains(logContent, "permission denied") {
+				result.IsWarn = true
+				result.Message = fmt.Sprintf("Found 'permission denied' in node-agent* pod log file(s)")
+				results = append(results, result)
+				continue
+			}
+
+			if strings.Contains(logContent, "error") || strings.Contains(logContent, "panic") || strings.Contains(logContent, "fatal") {
+				result.IsWarn = true
+				result.Message = fmt.Sprintf("Found error|panic|fatal in node-agent* pod log file(s)")
+				results = append(results, result)
+			}
+		}
+
+		results = append(results, &AnalyzeResult{
+			Title:   "Velero Logs analysis",
+			IsPass:  true,
+			Message: fmt.Sprintf("Found %d log files", len(logs)),
+		})
+	}
+	return results
+}
+
+func aggregateResults(results []*AnalyzeResult) []*AnalyzeResult {
+	out := []*AnalyzeResult{}
+	resultFailed := false
+	for _, result := range results {
+		if result.IsFail {
+			resultFailed = true
+		}
+		out = append(out, result)
+	}
+	if len(results) > 0 {
+		if resultFailed == false {
+			out = append(out, &AnalyzeResult{
+				Title:   "Velero Status",
+				IsPass:  true,
+				Message: "Velero setup is healthy",
+			})
+		}
+		if resultFailed == true {
+			out = append(out, &AnalyzeResult{
+				Title:   "Velero Status",
+				IsWarn:  true,
+				Message: "Velero setup is not entirely healthy",
+			})
+		}
+	}
+
+	return out
+}
+
+func GetVeleroBackupsDirectory(namespace string) string {
+	return fmt.Sprintf("cluster-resources/custom-resources/backups.velero.io")
```

Comment:
NIT: `fmt.Sprintf` is not necessary here and in the reset of the `GetXXX` functions below

Reply:
Courtesy of mass editing in vscode and leftovers from collector code 😀